### PR TITLE
Documentation for Timer class and cog

### PR DESCRIPTION
**Pull request by syed-safwaan**
Please leave review comments for the documentation, as well as for all the functionality of the Timer class if possible. thanks

### CODE REVIEW

File: cogs/cog_ticket.py [hunk 1]
```diff
@@ -29,32 +45,58 @@ async def setup_tickets(self, ctx):
...
+                                                                                    id=category_id))
+
+        # Set the permissions of the channel such that only the requesting user can see it
         await ticket_channel.set_permissions(ctx.author, read_messages=True, send_messages=False, embed_links=True,
                                              attach_files=True)
 
+        # Continue setup in the ticket channel by prompting user for their question, delete everything if timed out
         q_prompt = await ticket_channel.send(ctx.author.mention, embed=discord.Embed(
             description="Type in your question here.",
             colour=cfg.colors.WSU_GOLD
```
Comment (AdiSai) [inline cogs/cog_ticket.py:102]: It would be nice to have a prompt Embed factory too

File: cogs/cog_timer.py [hunk 2]
```diff
@@ -13,13 +12,33 @@ def __init__(self, bot):
...
+            await ctx.send(embed=discord.Embed(
+                description="Timer duration not in acceptable range! [1 .. 120]",
+                colour=cfg.colors.ERROR
+            ))
+            return
 
         # Timer creation
         msg = ' '.join(msg)
-        timer_obj = timer.Timer(userid=userid, td_secs=int(time) * 60, msg=msg, discord_message=ctx.message)
+        timer_obj = timer.Timer(userid=user.id, td_secs=int(time) * 60, msg=msg, discord_message=ctx.message)
```
Comment (AdiSai) [inline cogs/cog_timer.py:41]: Redundant cast: we've already casted time to an int, but now we're doing it again

File: cogs/cog_timer.py [hunk 3]
```diff
@@ -29,29 +48,35 @@ async def set_timer(self, ctx, time: str, *msg: str):
...
+        """
 
-        # Identify users and fetch their timers
-        userid = ctx.author.id
+        # Identify user and fetch their timers
+        user = ctx.author
         user_timers = sorted(
-            [obj for obj in timer_priority_queue.TimerPriorityQueue.get_instance().user_map[userid] if type(obj) == timer.Timer]
+            [obj for obj in timer_priority_queue.TimerPriorityQueue.get_instance().user_map.get(user.id, []) if
+             type(obj) == timer.Timer]
```
Comment (AdiSai) [inline cogs/cog_timer.py:63]: I can see why you used .get here, but you could check user_map earlier to ensure the list isn't empty

File: main.py [hunk 4]
```diff
@@ -70,13 +84,14 @@ async def on_command_error(ctx, error):
...
 ]
 
 if __name__ == '__main__':  # Ensures this is the file being ran
     for extension in extensions:
         bot.load_extension(extension)  # Loads every extension.
 
 keep_alive()  # Starts a webserver to be pinged.
+# load_dotenv()
+# token = os.getenv("DISCORD_BOT_SECRET")
 token = os.environ.get("DISCORD_BOT_SECRET")
```
Comment (AdiSai) [inline main.py:96]: os.environ["DISCORD_BOT_SECRET"] is another feisable solution

File: main.py [hunk 5]
```diff
@@ -57,9 +74,6 @@ async def on_ready():  # When the bot is ready
     print(bot.user)  # Prints the bot's username and identifier
     await async_tasks.run_tasks()
 
-@bot.event
-async def on_command_error(ctx, error):
-    await ctx.send(f'Error! {error}')
```
Comment (AdiSai) [inline main.py:62]: I think @MSaimBeg would agree that this should be kept

File: main.py [hunk 6]
```diff
@@ -70,13 +84,14 @@ async def on_command_error(ctx, error):
     'cogs.cog_profile',
     'cogs.cog_ticket',
     'cogs.cog_todolist',
-    'cogs.cog_easteregg'
```
Comment (AdiSai) [inline main.py:73]: Shouldn't be removed

File: cogs/cog_timer.py [hunk 7]
```diff
@@ -61,95 +86,110 @@ async def list_timers(self, ctx, limit=None):
...
                 name=f"Expires At",
-                value=time_utils.utc_to_dest(t.end_time).strftime("%c"),
+                value=time_utils.utc_to_dest(t.end_time).strftime("%c"),  # expiration date
                 inline=True
-            ).add_field(
+            ).add_field(  # empty inline field to allow wrapping of each timer field group
                 name='\u200b',
                 value='\u200b',
                 inline=True
             )
```
Comment (AdiSai) [inline cogs/cog_timer.py:103]: Make this a constant in cfg so it can be used elsewhere

File: cogs/cog_timer.py [hunk 8]
```diff
@@ -61,95 +86,110 @@ async def list_timers(self, ctx, limit=None):
...
             icon_url=ctx.author.avatar_url
-        ).add_field(
+        ).add_field(  # message
             name="Message",
             value=t.msg,
             inline=False
-        ).add_field(
+        ).add_field(  # creation date
             name="Created at",
             value=time_utils.utc_to_dest(t.start_time).strftime("%c"),
```
Comment (AdiSai) [inline cogs/cog_timer.py:142]: ideally, we would also specify the destination tz based on the user's profile

**Review by AdiSai — CHANGES_REQUESTED** [review]
Everything looks good except for a few things in main that were accidentally removed

File: cogs/cog_ticket.py [hunk 9]
```diff
@@ -1,20 +1,36 @@
...
+
+        # Get server from the context of the command
         guild = ctx.guild
 
+        # If this server already has a ticket category, exit
+        if cfg.server_ticket_ctgs.get(guild.id) is not None:
+            await ctx.send(embed=discord.Embed(
+                description="This server already has tickets set up!",
+                colour=cfg.colors.ERROR
+            ))
```
[the remark this answers is not in the document]
Comment (syed-safwaan) [reply cogs/cog_ticket.py:30]: Sounds like a great idea, I'll get on that.

**Review by AdiSai — APPROVED** [review]
Good job!

**Pull request merged**